src/engine/sync/synccontrol.cpp

```diff
@@ -305,17 +306,23 @@ void SyncControl::updateTargetBeatDistance() {
     m_pBpmControl->setTargetBeatDistance(targetDistance);
 }
 
-double SyncControl::getBpm() const {
+mixxx::Bpm SyncControl::getBpm() const {
+    const auto bpm = mixxx::Bpm(m_pBpm->get());
     if (kLogger.traceEnabled()) {
         kLogger.trace() << getGroup() << "SyncControl::getBpm()"
-                        << m_pBpm->get() << "/" << m_leaderBpmAdjustFactor;
+                        << bpm << "/" << m_leaderBpmAdjustFactor;
+    }
+    if (!bpm.isValid()) {
+        return {};
```

Comment:
as elsewhere, rather than returning "empty" I would prefer to explicitly return the Invalid value

Reply:
That would require to introduce a named constant for *default constructed* instances of `Bpm`.